src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/Processors/LedgerTransactionMarkers/OriginLedgerTransactionMarkerProcessor.cs

```diff
@@ -91,16 +96,22 @@ public void VisitTransaction(CoreApiSdk.Model.CommittedTransaction committedTran
                 });
         }
 
-        if (committedTransaction.LedgerTransaction is CoreApiSdk.Model.UserLedgerTransaction)
+        var origin = committedTransaction.LedgerTransaction switch
         {
-            _ledgerTransactionMarkersToAdd.Add(
-                new OriginLedgerTransactionMarker
-                {
-                    Id = _context.Sequences.LedgerTransactionMarkerSequence++,
-                    StateVersion = stateVersion,
-                    OriginType = LedgerTransactionMarkerOriginType.User,
-                });
-        }
+            FlashLedgerTransaction => LedgerTransactionMarkerOriginType.Flash,
+            GenesisLedgerTransaction => LedgerTransactionMarkerOriginType.Genesis,
+            RoundUpdateLedgerTransaction => LedgerTransactionMarkerOriginType.RoundUpdate,
+            UserLedgerTransaction => LedgerTransactionMarkerOriginType.User,
```

Comment:
This is fine as-is, but conceptually, we can split a `LedgerTransaction` origin as `User`, `Validator` (round update) and `System` (genesis or protocol update = flash). I think this split is perhaps a little more instructive, but I don't feel super strongly if you feel this is much more consistent. Interested to hear your thoughts.